### PR TITLE
Update `validation` package

### PR DESCRIPTION
**Pull request by user27182**
### Overview
Update `validation` package. This is a follow-up PR from #5100.

- Rename package: #5558
- Simplify function names, e.g. `check_has_shape` -> `check_shape`
- Add array wrappers to process flat sequences and nested 2D sequences separately from all other array-like inputs. This allows sequence inputs to be processed _without_ needing to cast to a numpy array first.
- Change and update definitions for checking `scalar` and `number` types
- Add `check_numeric` function
- Update package typing:
   - use numpy types `ArrayLike`, `DTypeLike`, etc. 
   - use updated generic PyVista types, e.g.`Vector[float]` from #5171 and #5285.
- Rework `check_string_in_iterable` as `check_contains` for general use with any input type and any container.
- Remove functions `check_is_dytpe`, `validate_dytpe` and `check_is_arraylike` since their use cases are very limited and type errors are already handled by NumPy.
- Remove `check_iterable_of_strings`;  the more general `check_iterable_items` function should be used instead.

Package function names may also be renamed accordingly to better match the new pyvista type names (as applicable).

Enhacements for follow-up PRs:
- Add `optional` param to checkers to do nothing if input is `None`
- Add `check_ndim` to check array dimensionality
- Remove use of `inspect` module to improve performance

### CODE REVIEW

**Comment by tkoyama010** [comment]
pre-commit.ci autofix

File: pyvista/core/input_validation/validate.py [hunk 1]
```diff
@@ -893,8 +898,8 @@ def validate_arrayN_uintlike(arr, /, *, reshape=True, **kwargs):
     Validate a 1D array with four non-negative integer-like elements.
 
     >>> import numpy as np
-    >>> import pyvista.core.input_validation as valid
-    >>> arr = valid.validate_arrayN_uintlike((1.0, 2.0, 3.0, 4.0))
+    >>> from pyvista.core import input_validation
+    >>> arr = validate.validate_arrayN_uintlike((1.0, 2.0, 3.0, 4.0))
```
Comment (tkoyama010) [inline pyvista/core/input_validation/validate.py:902]:
```suggestion
    >>> arr = input_validation.validate_arrayN_uintlike((1.0, 2.0, 3.0, 4.0))
```

**Comment by user27182** [comment]
FYI I've decided to update the typing first (this PR), along with some small updates, and make changes to the package name and structure in a separate PR. 

This is because with the structural changes, the diff from main was impossible to review. It should be much easier to see the changes made by this PR now.

**Comment by tkoyama010** [comment]
> FYI I've decided to update the typing first (this PR), along with some small updates, and make changes to the package name and structure in a separate PR.
> 
> This is because with the structural changes, the diff from main was impossible to review. It should be much easier to see the changes made by this PR now.

Thank you. We are very much looking forward to getting this PR done.

**Comment by user27182** [comment]
Thank you for your patience 🙏

**Comment by user27182** [comment]
Update: The typing for `validate_array` is complex and highly dynamic. A broad number of input types are accepted, and with all the different optional parameters, the return type varies greatly. The typing updates I am working on will hopefully result in the output types from this function being as narrow as possible while maintaining flexibility with the input types.

**Pull request closed without merging**